features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/wrappers/BitmapWrapper.kt

```diff
@@ -27,4 +27,21 @@ internal class BitmapWrapper {
             null
         }
     }
+
+    internal fun createScaledBitmap(
+        src: Bitmap,
+        dstWidth: Int,
+        dstHeight: Int,
+        filter: Boolean
+    ): Bitmap? {
+        @Suppress("SwallowedException")
+        return try {
+            Bitmap.createScaledBitmap(src, dstWidth, dstHeight, filter)
+        } catch (e: IllegalArgumentException) {
+            // should never happen since config is given as valid type and width/height are
+            // normalized to be at least 1
+            // TODO: REPLAY-1364 Add logs here once the sdkLogger is added
```

Comment:
I can add the logger in this pr, but because the logger is not passed at the moment in base64 it will require a change to several classes that may be better to do separately (and if so, for all necessary places in base64)

Reply:
you do not have to change it now but for situations like this one we usually prefer having extension methods as:
```
fun Bitmap.safeCreateScaledBitmap(..)
```

Reply:
Specifically in this case, I'm not sure that its supported because these are static methods of the class. I see that [support for extending static methods](https://github.com/Kotlin/KEEP/issues/348) is still in the [proposal stage](https://github.com/Kotlin/KEEP/blob/statics/proposals/statics.md#static-extensions) for kotlin.